src/core/toga/__init__.py

```diff
@@ -100,4 +103,6 @@ def set_platform(module_name=None, local_vars=locals()):
             raise
 
 # On first import, do an autodetection of platform.
-set_platform()
+# Bypass this for docs.
+if not os.environ.get('IS_SPHINX'):
```

Comment:
this is a really ugly workaround, we shouldn't really have a "is documentation" step here, but the code gets executed whenever you import any of the toga modules, which autodoc doesn't like. Also it could cause weird import loops. 

What does this code do and how is the user expected to use it?